docs/waveform_plugin.rst

```diff
@@ -0,0 +1,91 @@
+.. _waveform_plugin:
+
+------------------------------------------------------
+Making new waveform approximants available to PyCBC
+------------------------------------------------------
+
+=================================================
+Adding a custom waveform model within a script
+=================================================
+
+By example, the following script shows how to write a waveform model
+in the form required for PyCBC. We can also make this new waveform directly
+accessible by using the `pycbc.waveform.plugin.add_custom_waveform` function.
+If you are developing in a notebook or self-contained script, this may be
+what you want to do. However, if you want to make your waveform available
+to pycbc-based executables such as PyCBC Inference, also read the next
+section. 
+
+Ther are two kinds of models you can make. In the example below, we
+make a time-domain model. You can also make a freuqency-domain model. The only
+difference is that your function should return pycbc FrequencySeries and
+the required sample step option is 'delta_f' instead of 'delta_t'. 
+
+Each waveform generation function must take only keyword arguments, and
+should be able to take an arbitrary number of them. You may add new parameters
+as you like. These will be automatically useable by PyCBC Inference and
+other pycbc codes.
+
+Each waveform model must have an associate 'approximant' name, which identifies
+the model and distinguishes it from any other. If the name has already been
+used, you may need to select a different name. 
+
+.. plot:: ../examples/waveform/add_waveform.py
+   :include-source:
+
+=================================================
+Creating a plugin for PyCBC
+=================================================
+
+To make a waveform model universally available to PyCBC so it can be called
+from PyCBC Inference, or the pycbc-based searched codes, you can create
+a plugin package which advertises your model. PyCBC will automatically
+detects your package and makes your waveform model available for use.
+
+The steps are:
+
+ * Create a waveform model just like as in the above example
+ * Create a python package for your module
+ * In your packages setup.py advertise that it contains a PyCBC compatible
+   waveform model in it's 'entry_points' option. 
+  
+The setup.py would like the following, the key addition being the 'entry_points'
+parameter passed to setup.py. 
```

Comment:
"The setup.py would like the following," -> "Your `setup.py` should look like the following,"
Also, 'entry_points' -> `entry_points`